lib/AppleNote.rb

```diff
@@ -383,20 +384,31 @@ def all_embedded_objects
     @embedded_objects + @embedded_objects_recursive
   end
 
+  ##
+  # Unique ID for the note — prefer UUID if available, fall back to database ID
+  def unique_id
+    return @unique_id if defined?(@unique_id)
+    @unique_id = uuid.empty? ? note_id : uuid
+  end
+
+  ##
+  # Generate a file name for exporting this note to an HTML file
+  def title_as_filename(ext = '')
+    "#{unique_id} - #{title.tr('/:', '_')}#{ext}"
```

Comment:
This errors on my test database which has some deleted notes in it. At least one ended up with the title field zero'd out, so this needs to check if title exists before calling tr on it.